services/controlplane/client/client_test.go

```diff
@@ -0,0 +1,329 @@
+	"github.com/rudderlabs/rudder-server/services/controlplane/identity"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSendFeatures(t *testing.T) {
+	validSecret := "valid-secret"
+
+	wantBody := `{"components":[{"name":"test","features":["feature1","feature2"]}]}`
+
+	testcases := []struct {
+		name     string
+		identity identity.Identifier
+
+		wantPath string
+		wantErr  error
+	}{
+		{
+			name: "workspace",
+			identity: &identity.Workspace{
+				WorkspaceID:    "valid-workspace-id",
+				WorkspaceToken: validSecret,
+			},
+
+			wantPath: "/data-plane/v1/workspaces/valid-workspace-id/settings",
+			wantErr:  nil,
+		},
+		{
+			name: "namespace",
+			identity: &identity.Namespace{
+				Namespace:    "valid-namespace",
+				HostedSecret: validSecret,
+			},
+			wantPath: "/data-plane/v1/namespaces/valid-namespace/settings",
+			wantErr:  nil,
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				u, _, _ := r.BasicAuth()
+
+				require.Equal(t, "application/json", r.Header.Get("Content-Type"))
+				require.Contains(t, r.UserAgent(), "control-plane/features")
+				require.Equal(t, validSecret, u)
+				require.Equal(t, http.MethodPost, r.Method)
+				require.Equal(t, tc.wantPath, r.URL.Path)
+
+				body, err := io.ReadAll(r.Body)
+				require.NoError(t, err)
+				require.Equal(t, wantBody, string(body))
+
+				w.WriteHeader(http.StatusNoContent)
+			}))
+			defer s.Close()
+
+			c := client.NewClient(s.URL, tc.identity, client.WithHTTPClient(s.Client()))
+
+			err := c.SendFeatures(context.Background(), "test", []string{"feature1", "feature2"})
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestDestinationHistory(t *testing.T) {
+	validSecret := "valid-secret"
+
+	body, err := os.ReadFile("./testdata/destination_history.json")
+	require.NoError(t, err)
+
+	expectedDestination := backendconfig.DestinationT{
+		ID:   "2ENkYVMMUInzUGYp32R6dVghgvj",
+		Name: "webhook.site",
+		DestinationDefinition: backendconfig.DestinationDefinitionT{
+			ID:          "1aIXpUrvpGno4gEuF2GvI3O9dOe",
+			Name:        "WEBHOOK",
+			DisplayName: "Webhook",
+			Config: map[string]interface{}{
+				"transformAt":             "processor",
+				"transformAtV1":           "processor",
+				"saveDestinationResponse": false,
+			},
+			ResponseRules: map[string]interface{}(nil),
+		},
+		Config: map[string]interface{}{
+			"webhookMethod": "POST",
+			"webhookUrl":    "https://webhook.site/4e257da5-000e-4390-9ad2-d28d37fa934e",
+		},
+		Enabled:            true,
+		WorkspaceID:        "1wBceIM3I4Fg5PJYiq8tbbvm3Ww",
+		Transformations:    []backendconfig.TransformationT(nil),
+		IsProcessorEnabled: false,
+		RevisionID:         "2ENkYQ1MR9f83YhqS5k7uTSe5XH",
+	}
+
+	testcases := []struct {
+		name       string
+		identity   identity.Identifier
+		region     string
+		revisionID string
+
+		responseBody   string
+		responseStatus int
```

Comment:
We are not using this in any test case. Are we planning to use this for later purposes?